src/xenia/kernel/xsession.h

```diff
@@ -337,10 +389,19 @@ class XSession : public XObject {
                                     uint32_t properties_ptr,
                                     XSESSION_SEARCHRESULT* result);
 
-  bool is_session_created_ = false;
-  XSESSION_STATE session_state = XSESSION_STATE::LOBBY;
+  // uint64_t migrated_session_id_;
   uint64_t session_id_;
 
+  bool is_created_ = false;
+  bool is_migrated_ = false;
+  bool is_deleted_ = false;
+  bool is_host_ = false;
```

Comment:
Maybe instead of creating 4 separate entries you can handle it by a bitmask an explicit values from enum class